pycnn.py

```diff
@@ -41,16 +41,45 @@
 
 
 class pycnn(object):
+    """Image Processing with Cellular Neural Networks (CNN).
+
+    Cellular Neural Networks (CNN) are a parallel computing paradigm that was
+    first proposed in 1988. Cellular neural networks are similar to neural
+    networks, with the difference that communication is allowed only between
+    neighboring units. Image Processing is one of its applications. CNN
+    processors were designed to perform image processing; specifically, the
+    original application of CNN processors was to perform real-time ultra-high
+    frame-rate (>10,000 frame/s) processing unachievable by digital processors.
+
+    This python library is the implementation of CNN for the application of
+    Image Processing.
+
+
+    Attributes:
+        n (int): The number of rows from the image matrix.
+        m (int): The number of columns form the image matrix.
+    """
 
     def f(self, x, t, Ib, Bu, tempA):
+        """Computes the derivative of x at t."""
         x = x.reshape((self.n, self.m))
         dx = -x + Ib + Bu + sig.convolve2d(self.cnn(x), tempA, 'same')
         return dx.reshape(self.m * self.n)
 
     def cnn(self, x):
+        """Approximate sigmoidal kernel function for the neural network."""
```

Comment:
This function is missing arguments description.
x is input to the sigmoidal kernel function.